frontend/src/components/dialogs/CreateTerminalSessionDialog.vue

```diff
@@ -0,0 +1,266 @@
+              color="cyan darken-2"
+              active-class="g-border"
+              multiple
+            >
+              <terminal-shortcuts
+                :shoot-item="shootItem"
+                popper-boundaries-selector="#shortcut-tab"
+                @addTerminalShortcut="onAddTerminalShortcut"
+              ></terminal-shortcuts>
+            </v-list-item-group>
+          </v-list>
+        </v-tab-item>
+      </v-tabs-items>
+      <unverified-terminal-shortcuts-dialog
+        ref="unverified"
+      ></unverified-terminal-shortcuts-dialog>
+    </template>
+  </g-dialog>
+</template>
+
+<script>
+import GDialog from '@/components/dialogs/GDialog'
+import TerminalSettings from '@/components/TerminalSettings'
+import TerminalTarget from '@/components/TerminalTarget'
+import TerminalShortcuts from '@/components/TerminalShortcuts'
+import UnverifiedTerminalShortcutsDialog from '@/components/dialogs/UnverifiedTerminalShortcutsDialog'
+import { mapGetters } from 'vuex'
+import { terminalConfig } from '@/utils/api'
+import filter from 'lodash/filter'
+import isEmpty from 'lodash/isEmpty'
+import pick from 'lodash/pick'
+
+export default {
+  components: {
+    GDialog,
+    TerminalSettings,
+    TerminalTarget,
+    TerminalShortcuts,
+    UnverifiedTerminalShortcutsDialog
+  },
+  props: {
+    name: {
+      type: String
+    },
+    namespace: {
+      type: String
+    }
+  },
+  data () {
+    return {
+      tab: undefined,
+      targetTab: {
+        selectedTarget: undefined,
+        terminalTargetValid: false,
+        value: [],
+        initializedForTarget: undefined,
+        selectedConfig: undefined,
+        validSettings: true // settings are assumed to be valid initially as the defaults apply. Once the settings expansion panel is expanded, this property get's updated
+      },
+      shortcutTab: {
+        selectedShortcuts: undefined
+      }
+    }
+  },
+  computed: {
+    ...mapGetters([
+      'hasControlPlaneTerminalAccess',
+      'hasShootTerminalAccess',
+      'hasGardenTerminalAccess',
+      'isAdmin',
+      'shootByNamespaceAndName'
+    ]),
+    shootItem () {
+      return this.shootByNamespaceAndName({ name: this.name, namespace: this.namespace })
+    },
+    valid () {
+      switch (this.tab) {
+        case 'target-tab': {
+          if (!this.targetTab.terminalTargetValid) {
+            return false
+          }
+
+          return this.targetTab.validSettings
+        }
+        case 'shortcut-tab': {
+          return !isEmpty(this.shortcutTab.selectedShortcuts)
+        }
+        default: {
+          return false
+        }
+      }
+    },
+    isSettingsExpanded () {
+      return this.targetTab.value === 0
+    }
+  },
+  methods: {
+    async promptForSelections (initialState) {
+      this.initialize(initialState)
+      const confirmed = await this.$refs.gDialog.confirmWithDialog({
+        confirmCallback: async () => {
```

Comment:
It is not a callback. A callback is a function with two argument. First an error on errors, second is the returned value. Here we return a promise.